### PR TITLE
RepeatAttribute increments CurrentRepeatCount on each repeat iteration

### LINKED ISSUE #3662: TestContext.CurrentContext.CurrentRepeatCount only contains retry count not the repeat count
While implementing a test which uses Retry and Repeat attributes, I noticed that the property "TestContext.CurrentContext.CurrentRepeatCount" does only count retries but not repeats.

As the name of the property leads to the assumption that repeats will be counted, I open this is issue as an improvement.

### PR DESCRIPTION
**Pull request by Antash**
Closes #3662 
RepeatAttribute should be incrementing the test context CurrentRepeatCount property for the sake of consistency.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/nunit/nunit?pullRequest=3696) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/nunit/nunit?pullRequest=3696) before we can accept your contribution.<br/><br/>:x: Antash [sign now](https://cla.dotnetfoundation.org/nunit/nunit?pullRequest=3696)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/nunit/nunit?pullRequest=3696) it.</sub>

**Comment by Antash** [comment]
#3662

**Comment by Antash** [comment]
There is an CI issue although all tests are green.

**Comment by mikkelbu** [comment]
@Antash I've just retriggered builds on master and there we also fail with the same error.

Some googling indicates that setting `GenerateErrorForMissingTargetingPacks` to `false` would correct this, see dotnet/runtime#45755, or perhaps setting `DisableImplicitFrameworkReferences` to `true`, but I need to examine this in more detail. I'm unsure how this could start to fail without any changes on master, but I guess it must be a change in the CI or the packages that we depend on (or recursively depend on).

**Comment by Antash** [comment]
@mikkelbu thank you for the investigation! Hope someone from the team will fix this soon, don't feel confident enough to change anything there on my own.

**Review by mikkelbu — COMMENTED** [review]
I have PR that should solve the CI problem (even though I still not sure exactly what the problem is - which is very annoying).

Regarding this PR then I've left some comments, but the implementation seems sound. Besides this I also think we should update nunit/src/NUnitFramework/framework/TestContext.cs as it currently contains a TODO and some comments which are no longer correct (I don't know why the initial implementation left this part undone given the property name "CurrentRepeatCount" - but I cannot find any mention of the TODO in neither the issue nunit/nunit#2647 nor the resolving PR nunit/nunit#2648) 

https://github.com/nunit/nunit/blob/ad49f27294bd0f2677d8699756c6ccb10df600f8/src/NUnitFramework/framework/TestContext.cs#L173-L181

File: src/NUnitFramework/tests/Attributes/RepeatAttributeTests.cs [hunk 1]
```diff
@@ -67,6 +67,26 @@ public void RepeatWorksAsExpected(Type fixtureType, string outcome, int nTries)
             Assert.AreEqual(nTries, fixture.Count);
         }
 
+        [Test]
+        public void RepeatUpdatesCurrentRepeatCountPropertyOnEachAttempt()
+        {
+            RepeatingTestsFixtureBase fixture = (RepeatingTestsFixtureBase)Reflect.Construct(typeof(RepeatedTestVerifyAttempt));
+            ITestResult result = TestBuilder.RunTestCase(fixture, "PassesTwoTimes");
```
[the remark this answers is not in the document]
Comment (Antash) [reply src/NUnitFramework/tests/Attributes/RepeatAttributeTests.cs]: Done. No Idea why I decided to ignore nameof. May be I was looking into existing RetryAttributeTests which uses plane strings either.

**Comment by rprouse** [comment]
@Antash like the other PR, I've submitted a PR to your fork to fix the build. https://github.com/Antash/nunit/pull/2

**Review by rprouse — APPROVED** [review]
We just need to get the build passing. Thanks.

**Comment by Antash** [comment]
@rprouse thank you! All green now.

**Pull request merged**